tests/libs/test_matrix.py

```diff
@@ -47,3 +56,60 @@ def test_message_from_dict(token_network, get_accounts, get_private_key):
         message_from_dict(message_json)
 
     assert "Invalid message data. Can not find the data type" in str(excinfo.value)
+
+
+def test_deserialize_messages_empty():
+    messages = deserialize_messages(data="", peer_address=PEER_ADDRESS)
+    assert len(messages) == 0
+
+
+def test_deserialize_messages_empty_lines():
+    messages = deserialize_messages(data=" \r\n ", peer_address=PEER_ADDRESS)
+    assert len(messages) == 0
+
+
+def test_deserialize_messages_invalid_json():
+    messages = deserialize_messages(data="@@@", peer_address=PEER_ADDRESS)
```

Comment:
You could combine these three tests and use
```python
for data in ['', ' \r\n ', '@@@']:
    ...
```
Mostly a matter of taste, though.